test/SerializerTest.cpp

```diff
@@ -559,3 +564,260 @@ TEST(Serializer, serializeOptional) {
   EXPECT_THAT(sExpected, ::testing::Optional(std::string("hallo")));
   EXPECT_EQ(nilExpected, std::nullopt);
 }
+
+// _____________________________________________________________________________
+// Tests for CompressedSerializer
+// _____________________________________________________________________________
+
+// Simple identity "compression" for testing (no actual compression)
+auto identityCompress = [](ql::span<const char> data) {
+  return std::vector<char>(data.begin(), data.end());
+};
+auto identityDecompress = [](ql::span<const char> data, size_t) {
+  return std::vector<char>(data.begin(), data.end());
+};
+
+TEST(CompressedSerializer, Concepts) {
+  using Writer = CompressedWriteSerializer<ByteBufferWriteSerializer,
+                                           decltype(identityCompress)>;
+  using Reader = CompressedReadSerializer<ByteBufferReadSerializer,
+                                          decltype(identityDecompress)>;
+  static_assert(WriteSerializer<Writer>);
+  static_assert(!ReadSerializer<Writer>);
+  static_assert(ReadSerializer<Reader>);
+  static_assert(!WriteSerializer<Reader>);
+}
+
+TEST(CompressedSerializer, SimpleRoundtrip) {
+  auto blockSize = ad_utility::MemorySize::bytes(64);
```

Comment:
Decrease the blocksize even further, s.t. even this test case spans multiple blocks.